tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_bluetooth.py

```diff
@@ -27,15 +20,29 @@ def test_toggle_bluetooth_settings(self):
         settings = Settings(self.marionette)
         settings.launch()
 
+        self.assertFalse(self.data_layer.bluetooth_is_enabled)
+        self.assertFalse(self.data_layer.get_setting('bluetooth.visible'))
+
         bluetooth_settings = settings.open_bluetooth_settings()
         bluetooth_settings.enable_bluetooth()
 
+        self.assertTrue(self.data_layer.bluetooth_is_enabled)
+
         bluetooth_settings.tap_rename_my_device()
         bluetooth_settings.type_phone_name(device_name)
         bluetooth_settings.tap_update_device_name_ok()
 
         bluetooth_settings.enable_visible_to_all()
 
-        # Now have host machine inquire and shouldn't find our device
-        device_found = self.bluetooth_host.is_device_visible(device_name)
-        self.assertTrue(device_found, "Host should see our device (device discoverable mode is ON)")
+        self.assertTrue(self.data_layer.get_setting('bluetooth.visible'))
+
+        self.marionette.switch_to_frame()
+        discoverable = self.marionette.execute_script("""
+            return window.wrappedJSObject.Bluetooth.defaultAdapter.discoverable;
+        """)
```

Comment:
Oops. For some reason my comment did not submit. Instead of using `"""` just use single speech marks:
```python
discoverable = self.marionette.execute_script(
    "return window.wrappedJSObject.Bluetooth.defaultAdapter.discoverable;")
```